### ISSUE TITLE
FPS rate decreases continuously after playing a GBC game for a while.

### ISSUE THREAD

**Issue by KingMucus** [visualboyadvance-m/visualboyadvance-m#111]
### Expected behavior

Constant 60 FPS at all times

### Actual behavior

After 10 or 15 minutes playing, the FPS rate decreases to below the ideal 60 rate and keeps diminishing until it stays at 30 fps or so and then I close the emu and open it again.

### Steps to reproduce the behavior

Open a GBC game and play for a while.

### ROM details

Pokémon Kristall (German Pokémon Crystal)

Which games are affected by the issue (please state if GBA or GB game):

It is a GBC game.

Which games, if any, are NOT affected by the issue (please state if GBA or GB game):

### Option details

Video driver (simple, opengl or cairo):

I tried both opengl and simple. Affects both.

Sound driver (directsound, xaudio, openal, SDL):

SDL.

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.):

Fedora Linux

If using Linux, specify if using xorg or Wayland:

I'm using xorg because I'm running the KDE spin.

Version of code (pre-built binary version, or commit ref, or just "master"):

The one availabe from rpmfusion

### Hardware details:

If your joystick or gamepad isn't working in Linux, please try to troubleshoot
using the information on this page:
https://wiki.archlinux.org/index.php/Gamepad **BEFORE** filing an issue. We use
SDL for joystick support, any problems will generally be an issue with SDL and
**NOT** the emulator.

If relevant, the joystick you are using and the drivers you are using for it:

**Comment by ZachBacon** [comment]
Can you provide the URL for the rpm fusion build? Chances are it maybe out of date with what we have available.

**Comment by KingMucus** [comment]
The about dialog states the version is 2.0.0 Beta3-07032017.

The rpm packages installed through "dnf install vbam-gtk" is here: https://www.rpmfind.net/linux/RPM/rpmfusion/free/fedora/25/armhfp/Packages/v/vbam-gtk-1.8.0.1229-3.fc25.armv7hl.html

That command pulls the vbam-common and vbam-sdl packages as dependencies, among others.

**Comment by ZachBacon** [comment]
Yeah that's old, we don't even support the gtk based interface anymore. We only support the command line sdl binary, the libretro interface and the wxwidgets interface for a better cross platform experience. If you want I suggest compiling from git, once you clone it, you can run the installdeps.sh file to install the needed libraries from fedora's repository and then all you need to do is follow the instructions it provides afterwards

**Comment by KingMucus** [comment]
I've also compiled it from GIT a acouple of days ago. The about dialog says the GIT commit version is b59b7b394cff939c2afcc59a3390a62b32355b52.

I understand that the GTK version is not supported anymore, but I believe it does not affect much the underlying functioning.

**Comment by rkitover** [comment]
@KingMucus you don't need to use rpmfusion, there are official Fedora packages in the Fedora repos, the package is called visualboyadvance-m:

https://apps.fedoraproject.org/packages/visualboyadvance-m

**Comment by Calmarius** [comment]
Compiled the latest git. Memory usage goes up gradually too. My guess is it's allocating stuff constantly putting ever incrasing stuff into an array and looping through it constantly... Within 10 minutes it gets predictably slower always the music becomes choppier and gradually turns the whole thing into a slideshow.

**Comment by noabody** [comment]
I have the same problem.  Libretro is great but tailoring cores for it seems to result in the inevitable loss of functionality that I desire (cheats, shaders, etc.).  If there was a way to seamlessly integrate the core unique features into the GUI with no such loss, I would not even attempt to build from source.

Below is my full report but it doesn't indicate why the process starts at 15% CPU utilization and gradually increases until the point where (at 25%) the sound and game-play are adversely affected to where the user must shut the program down and restart.

My two cents worth about vba-m:  Best shaders, best features, bar none.  Maybe the mgba/medusa/higan types will get there someday but this project has pretty much everything right now.

[report.txt](https://github.com/visualboyadvance-m/visualboyadvance-m/files/1185057/report.txt)

**Comment by retro-wertz** [comment]
i see "decrease in framerate" after a while running, and Linux... so, 

this is related to the Refresh() vs dcClient() thingie.. this should be fixed already in latest commit...
